v3/src/components/graph/models/data-configuration-model.ts

```diff
@@ -93,6 +93,21 @@ export const DataConfigurationModel = types
       return selection.filter(caseId => self.filteredCases?.hasCaseId(caseId))
     }
   }))
+  .views(self => ({
+    attributeValuesForPlace(place:GraphAttrPlace):string[] {
+      const attrID = self.attributeID(place),
+        dataset = self.dataset
+      return attrID ? self.cases.map(anID => String(dataset?.getValue(anID, attrID))) : []
+    },
+    categorySetForPlace(place:GraphAttrPlace):Set<string> {
+      const result:Set<string> = new Set(this.attributeValuesForPlace(place))
+      result.delete('')
+      if(result.size === 0) {
+        result.add('__main__')
+      }
+      return result
+    }
```

Comment:
There should be some tests of these methods as well, but I leave it to your discretion whether that's a task for this PR or not.

Reply:
I'll include tests in this PR. Also, I think the detection/elimination of empty values should occur in attributeValuesForPlace, obviating the need to delete them from the Set.

Reply:
I've added them now.